src/churn/churn.ts

```diff
@@ -351,9 +350,15 @@ var log :logging.Log = new logging.Log('churn');
       //       'ciphertext_dfa': regex2dfa('^.*$'),
       //       'ciphertext_max_len': 1450
       //     }
-      this.pipe_.bindLocal(natEndpoints.internal);
-      this.pipe_.setBrowserEndpoint(webRtcEndpoint);
-      this.pipe_.bindRemote(remoteEndpoint).then(this.haveForwardingSocketEndpoint_);
+      Promise.all<any>([
+          this.pipe_.bindLocal(natEndpoints.internal),
+          this.pipe_.setBrowserEndpoint(webRtcEndpoint),
+          this.pipe_.bindRemote(remoteEndpoint)]).then((answers:any[]) => {
```

Comment:
Technically, this could still be slower than the old implementation, because `this.haveForwardingSocketEndpoint_` now waits for all three promises, even though it only depends on one of them.  How about

```typescript
var bindRemote = this.pipe_.bindRemote(remoteEndpoint).then((mirrorEndpoint:net.Endpoint) : net.Endpoint => {
  this.haveForwardingSocketEndpoint_(mirrorEndpoint);
  return mirrorEndpoint;
});

Promise.all<any>([
    this.pipe_.bindLocal(natEndpoints.internal),
    this.pipe_.setBrowserEndpoint(webRtcEndpoint),
    bindRemote
])...
```

This would also avoid the need for a cast below.